cardano-node/src/Cardano/Common/Protocol.hs

```diff
@@ -53,44 +55,54 @@ data Protocol =
 
 
 data SomeProtocol where
-  SomeProtocol :: (RunDemo blk, TraceConstraints blk)
+  SomeProtocol :: (Demo.RunDemo blk, TraceConstraints blk)
                => Consensus.Protocol blk -> SomeProtocol
 
 fromProtocol :: CardanoConfiguration -> Protocol -> IO SomeProtocol
+
 fromProtocol _ ByronLegacy =
   error "Byron Legacy protocol is not implemented."
+
 fromProtocol _ BFT =
-    case Consensus.runProtocol p of
-      Dict -> return $ SomeProtocol p
+  case Consensus.runProtocol p of
+    Dict -> return $ SomeProtocol p
+
   where
     p = Consensus.ProtocolMockBFT defaultSecurityParam
+
 fromProtocol _ Praos =
-    case Consensus.runProtocol p of
-      Dict -> return $ SomeProtocol p
+  case Consensus.runProtocol p of
+    Dict -> return $ SomeProtocol p
+
   where
     p = Consensus.ProtocolMockPraos defaultDemoPraosParams
+
 fromProtocol _ MockPBFT =
-    case Consensus.runProtocol p of
-      Dict -> return $ SomeProtocol p
+  case Consensus.runProtocol p of
+    Dict -> return $ SomeProtocol p
+
   where
     p = Consensus.ProtocolMockPBFT defaultDemoPBftParams
-fromProtocol CardanoConfiguration{ccCore} RealPBFT = do
+
+fromProtocol cc RealPBFT = do
     let Core{ coGenesisFile
             , coGenesisHash
             , coPBftSigThd
-            } = ccCore
+            } = ccCore cc
         genHash = either (throw . ConfigurationError) identity $
                   decodeAbstractHash coGenesisHash
         cvtRNM :: RequireNetworkMagic -> RequiresNetworkMagic
         cvtRNM NoRequireNetworkMagic = RequiresNoMagic
         cvtRNM RequireNetworkMagic   = RequiresMagic
 
-    gcE <- runExceptT (Genesis.mkConfigFromFile (cvtRNM $ coRequiresNetworkMagic ccCore) coGenesisFile genHash)
+    gcE <- runExceptT (Genesis.mkConfigFromFile
```

Comment:
Ah these are going to disrupt changes I'm currently making here re: exception handling. That fine though, I'll deal with it in the rebase :).

Reply:
Dropping NFP/RWC was kind of a global change..  I hope you're not too angry. : -/

Reply:
Just the opposite in fact :slightly_smiling_face: Happy to see them go!